pkg/vdri/httpbinding/vdri.go

```diff
@@ -57,12 +65,103 @@ func (v *VDRI) Accept(method string) bool {
 
 // Store did doc
 func (v *VDRI) Store(doc *did.Doc, by *[]vdriapi.ModifiedBy) error {
-	return errors.New("not supported")
+	logger.Warnf(" store not supported in http binding vdri")
+	return nil
 }
 
 // Build did doc
 func (v *VDRI) Build(pubKey *vdriapi.PubKey, opts ...vdriapi.DocOpts) (*did.Doc, error) {
-	return nil, errors.New("not supported")
+	// Apply options
+	docOpts := &vdriapi.CreateDIDOpts{}
+
+	for _, opt := range opts {
+		opt(docOpts)
+	}
+
+	publicKey := did.PublicKey{
+		ID:         pubKeyIndex1,
+		Type:       pubKey.Type,
+		Controller: pubKeyController,
+		Value:      []byte(pubKey.Value),
+	}
+
+	t := time.Now()
+
+	didDoc := &did.Doc{
+		Context:   []string{did.Context},
+		PublicKey: []did.PublicKey{publicKey},
+		Created:   &t,
+		Updated:   &t,
+	}
+
+	if docOpts.ServiceType != "" {
+		s := did.Service{
+			ID:              svcEndpointIndex1,
+			Type:            docOpts.ServiceType,
+			ServiceEndpoint: docOpts.ServiceEndpoint,
+		}
+
+		if docOpts.ServiceType == vdriapi.DIDCommServiceType {
+			s.RecipientKeys = []string{publicKey.ID}
+			s.Priority = 0
+		}
+
+		didDoc.Service = []did.Service{s}
+	}
+
+	docBytes, err := didDoc.JSONBytes()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get document bytes : %s", err)
+	}
+
+	var reqBody io.Reader
+	if docOpts.RequestBuilder != nil {
+		reqBody, err = docOpts.RequestBuilder(docBytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build request : %s", err)
+		}
+	} else {
+		bytes.NewReader(docBytes)
+	}
+
+	resDoc, err := v.sendCreateRequest(reqBody)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send create DID request: %s", err)
+	}
+
+	return resDoc, nil
+}
+
+func (v *VDRI) sendCreateRequest(req io.Reader) (*did.Doc, error) {
+	httpReq, err := http.NewRequest(http.MethodPost, v.endpointURL, req)
```

Comment:
please leave a TODO for configuring timeouts properly on these external calls

Reply:
Agreed, will be addressed in #855 , added a TODO